### PR TITLE
Benchmark existing ways to check for `IDENTITY` CIDs

### PR DESCRIPTION
**Pull request by masih**
Benchmark existing ways to check for `IDENTITY` CIDs

Add benchmarks that compare two ways of checking for
`multihash.IDENTITY` code:
1. `Cid.Prefix().MhType`
2. Decode of `Cid.Hash()`

This benchmark illustrates that using Cid.Prefix is more efficient than
`multihash.Decode`. Users wishing to perform such a check should use
`Cid.Prefix`.

Consider that `Cid.Prefix` is already efficient enough and introducing a
dedicated API for performing this check will likely result in small
gains.

```
BenchmarkIdentityCheck
BenchmarkIdentityCheck/Prefix
BenchmarkIdentityCheck/Prefix-8         	351656472	         3.280 ns/op
BenchmarkIdentityCheck/MultihashDecode
BenchmarkIdentityCheck/MultihashDecode-8         	 5715622	       239.1 ns/op
```

Relates to #133

### CODE REVIEW

**Review by mvdan — COMMENTED** [review]
You can drop the `CidV1` part of the benchmark name; Cid is redundant since this module is go-cid, and v1 is redundant as we don't really care if the CID is v1 or not - it just so happens that all identity CIDs are currently v1. If they also supported a future CIDv2, then the benchmark would simply do both in each iteration.

You could also make the two methods sub-benchmarks, to simplify sharing code and make the output nicer. For example:

```
func BenchmarkIdentityCheck(b *testing.B) {
    // cid setup, ReportAllocs, etc
    b.Run("Prefix", func(b *testing.B) {
        b.RunParallel(func(pb *testing.PB) {
            // check via Prefix
        })
    })
    b.Run("MultihashDecode", func(b *testing.B) {
        b.RunParallel(func(pb *testing.PB) {
            // check via MultihashDecode
        })
    })
}
```

File: benchmark_test.go
Comment (mvdan) [inline benchmark_test.go:21]: I wouldn't name the inner one `bb`, because then it's too easy to make the mistake of calling methods on the wrong (parent) `b`, just like you did on line 24. If this sub-benchmark fails, only it should fail, not the entire parent benchmark.

File: benchmark_test.go
Comment (mvdan) [inline benchmark_test.go:62]: consider inlining generateCidV1, or even both funcs, now that they are only called in one place.

File: benchmark_test.go
Comment (mvdan) [inline benchmark_test.go:7]: keep the std imports separate :)

**Review by mvdan — APPROVED** [review]
SGTM! Probably wait for a review from a steward though.

File: benchmark_test.go
Comment (mvdan) [reply benchmark_test.go:21]: This is one example where shadowing is a good thing: it prevents using the wrong "level".

**Review by warpfork — APPROVED** [review]
I have no reason to complain about more benchmarks :)

If you still have the context in your head: I wouldn't mind a short paragraph of comment attached to the function about what you were looking for / why it mattered / and what the rough findings were at the time you committed.  (It'll get stale, sure -- but that doesn't mean it won't still be useful context clues to some future archaeologist^W reviewer.)

**Comment by masih** [comment]
Will do 🍻 

Context captured [here](https://github.com/ipfs/go-cid/issues/133)

TLDR; there are multiple ways of inferring if a CID is `IDENTITY` and one of them is more efficient. The one that's more efficient is efficient enough to avoid the need to implement new APIs to do this (see [closed PR](https://github.com/ipfs/go-cid/pull/134) for more details)

**Pull request merged**
